### PR TITLE
Enhancement: Boxes Indexer Integration test

### PR DESCRIPTION
**Pull request by ahangsu**

### CODE REVIEW

File: .test-env [hunk 1]
```diff
@@ -1,6 +1,6 @@
 # Configs for testing repo download:
 SDK_TESTING_URL="https://github.com/algorand/algorand-sdk-testing"
-SDK_TESTING_BRANCH="feature/box-storage"
+SDK_TESTING_BRANCH="ahangsu/indexer-integration-boxes"
```
Comment (ahangsu) [inline .test-env:3]: TODO: once algorand-sdk-testing PR is merged, do
```suggestion
SDK_TESTING_BRANCH="feature/box-storage"
```

File: tests/steps/other_v2_steps.py [hunk 2]
```diff
@@ -1023,6 +1023,12 @@ def algod_v2_client(context):
     context.app_acl = algod.AlgodClient(daemon_token, algod_address)
 
 
+@given("an indexer v2 client")
```
Comment (tzaffi) [inline tests/steps/other_v2_steps.py:1026]: 👶 - indexer is reborn!

Comment (tzaffi) [reply tests/steps/other_v2_steps.py:1026]: nit: might be nice to put right next to where `algod` is defined.

**Review by tzaffi — COMMENTED** [review]
Very tempted to approve, but put up some small nits worth discussing.

File: tests/steps/application_v2_steps.py [hunk 3]
```diff
@@ -1157,13 +1173,79 @@ def check_box_contents(
...
+    ), f"Expected box names array length does not match actual array length {(expected_box_names)} != {(actual_box_names)}"
+    assert set(expected_box_names) == set(
+        actual_box_names
+    ), f"Expected box names array does not match actual array {expected_box_names} != {actual_box_names}"
+
+
+@then(
+    'according to indexer, by parameter max {limit} and next "{next_page:MaybeString}", the current application should have the following boxes "{box_names:MaybeString}".'
+)
+def check_all_boxes_by_indexer(
```
Comment (tzaffi) [inline tests/steps/application_v2_steps.py:1234]: Is this step made obsolete by `check_all_boxes()` since it handles `indexer` as well?

Comment (tzaffi) [reply tests/steps/application_v2_steps.py:1234]: no it's not because of the `next` parameter.

Resolving

**Review by tzaffi — APPROVED** [review]
LGTM

File: tests/steps/application_v2_steps.py [hunk 4]
```diff
@@ -1173,7 +1255,14 @@ def check_all_boxes(context, box_names: str = None):
...
         actual_box_names
-    ), f"Expected box names array length does not match actual array length {(expected_box_names)} != {(box_response)}"
+    ), f"Expected box names array length does not match actual array length {(expected_box_names)} != {(actual_box_names)}"
     assert set(expected_box_names) == set(
         actual_box_names
     ), f"Expected box names array does not match actual array {expected_box_names} != {actual_box_names}"
+
+
+@then(
+    "I sleep for {millisecond_num} milliseconds for indexer to digest things down."
```
Comment (algochoi) [inline tests/steps/application_v2_steps.py:1265]: How about we move this to the other sleep function: https://github.com/algorand/py-algorand-sdk/blob/c0720e9dff101cf5dc2a476ad4820cae32c469d8/tests/steps/application_v2_steps.py#L422

File: tests/steps/steps.py [hunk 5]
```diff
@@ -39,6 +39,7 @@ def parse_string(text):
 
 
 token = "aaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaa"
+indexer_port = 59999
```
Comment (algochoi) [inline tests/steps/steps.py:42]: nit: if unused, can we delete?

File: tests/steps/application_v2_steps.py [hunk 6]
```diff
@@ -1132,16 +1136,28 @@ def check_found_method_or_error(context, methodsig, error: str = None):
...
+        if from_client == "algod":
+            box_response = context.app_acl.application_box_by_name(
+                context.current_application_id, box_name
+            )
+        elif from_client == "indexer":
+            box_response = context.app_icl.application_box_by_name(
+                context.current_application_id, box_name
+            )
+        else:
+            assert False, "expecting algod or indexer, get " + from_client
```
Comment (algochoi) [inline tests/steps/application_v2_steps.py:1159]: nit to use f-strs :
```suggestion
            assert False, f"expecting algod or indexer, got: {from_client}"
```

File: tests/steps/application_v2_steps.py [hunk 7]
```diff
@@ -1157,13 +1173,79 @@ def check_box_contents(
...
+            context.current_application_id, limit=limit_int
+        )
+    elif from_client == "indexer":
+        box_response = context.app_icl.application_boxes(
+            context.current_application_id, limit=limit_int
+        )
+    else:
+        assert False, "expecting algod or indexer, get " + from_client
+
+    assert expected_num_int == len(box_response["boxes"]), "expected box num"
```
Comment (algochoi) [inline tests/steps/application_v2_steps.py:1192]:
```suggestion
    assert expected_num_int == len(box_response["boxes"]), f"expected box num: {expected_num_int}, got {len(box_response["boxes"])}"
```

**Review by algochoi — COMMENTED** [review]
Minor formatting comments, looks good!

Non-urgent thought but I wonder if long term we aim to write indexer integration tests for all existing features and we want to split the indexer functionality into another file.

File: tests/steps/steps.py (hunk 5 above)
Comment (ahangsu) [reply tests/steps/steps.py:42]: this one is used in `other_v2_steps.py`

File: tests/steps/application_v2_steps.py (hunk 6 above)
Comment (ahangsu) [reply tests/steps/application_v2_steps.py]: oh boy, I don't know that we can actually use f-strs.

File: tests/steps/application_v2_steps.py (hunk 7 above)
Comment (ahangsu) [reply tests/steps/application_v2_steps.py:1192]: uh oh

File: tests/steps/application_v2_steps.py [hunk 8]
```diff
@@ -1157,13 +1173,81 @@ def check_box_contents(
...
+    if from_client == "algod":
+        box_response = context.app_acl.application_boxes(
+            context.current_application_id
+        )
+    elif from_client == "indexer":
+        box_response = context.app_icl.application_boxes(
+            context.current_application_id
+        )
+    else:
+        assert False, "expecting algod or indexer, get " + from_client
```
Comment (algochoi) [inline tests/steps/application_v2_steps.py:1221]:
```suggestion
        assert False, f"expecting algod or indexer, got: {from_client}"
```

File: tests/steps/application_v2_steps.py [hunk 9]
```diff
@@ -1157,13 +1173,81 @@ def check_box_contents(
...
+    if from_client == "algod":
+        box_response = context.app_acl.application_boxes(
+            context.current_application_id, limit=limit_int
+        )
+    elif from_client == "indexer":
+        box_response = context.app_icl.application_boxes(
+            context.current_application_id, limit=limit_int
+        )
+    else:
+        assert False, "expecting algod or indexer, get " + from_client
```
Comment (algochoi) [inline tests/steps/application_v2_steps.py:1195]:
```suggestion
        assert False, f"expecting algod or indexer, got: {from_client}"
```

File: tests/steps/steps.py (hunk 5 above)
Comment (algochoi) [reply tests/steps/steps.py:42]: Gotcha, thanks - we may have to delete the v1 `steps` so might have to port these over in the future, but looks fine as-is.

File: tests/steps/application_v2_steps.py (hunk 4 above)
Comment (ahangsu) [reply tests/steps/application_v2_steps.py]: yea I merged with a tricky way, hopefully it works out

Comment (ahangsu) [reply tests/steps/application_v2_steps.py]: okay seems the trick works

File: tests/steps/application_v2_steps.py [hunk 10]
```diff
@@ -415,14 +419,19 @@ def remember_app_id(context):
     context.app_ids.append(app_id)
 
 
-def wait_for_algod_transaction_processing_to_complete():
+@then(
+    "I sleep for {millisecond_num} milliseconds for indexer to digest things down."
```
Comment (algochoi) [inline tests/steps/application_v2_steps.py:423]: :+1:

**Review by algochoi — APPROVED** [review]

**Pull request merged**